fastify.js

```diff
@@ -399,7 +404,11 @@ function fastify (options) {
     // In the vast majority of cases, it's a network error and/or some
     // config issue on the the load balancer side.
     this.log.trace({ err }, 'client error')
-    socket.end(`HTTP/1.1 400 Bad Request\r\nContent-Length: ${body.length}\r\nContent-Type: application/json\r\n\r\n${body}`)
+
+    // If the socket is not writable, there is no reason to try to send data.
+    if (socket.writable) {
+      socket.end(`HTTP/1.1 400 Bad Request\r\nContent-Length: ${body.length}\r\nContent-Type: application/json\r\n\r\n${body}`)
```

Comment:
You can avoid generating the body

Reply:
I was thinking about making a supplemental check in case other lowlevel errors (other than connection reset) are raised with a non-writable socket. Do you think is over-engineering?